ParameterizedAssertionError should extend AssertionError

Old title was: "Use Assert in Theory, the testcae status is Error when assert fail, but I expected is failure(in  maven)"

code:

``` java
@RunWith(Theories.class)
public class TheoryTest {

    @DataPoints public static int[] numbers = {1, 2, 3};

    @Theory
    public void test(int number) {
        Assume.assumeThat(number, Is.is(2));
        Assert.assertThat(number * 10, Is.is(30));
    }

}
```

output(maven):

> Running TheoryTest
> Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.083 sec <<< FAILURE!
> test(TheoryTest)  Time elapsed: 0.022 sec  <<< ERROR!
> org.junit.experimental.theories.internal.ParameterizedAssertionError: test(ids[1])
>     at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
>     at org.junit.Assert.assertThat(Assert.java:865)
>     at org.junit.Assert.assertThat(Assert.java:832)
>     at TheoryTest.test(TheoryTest.java:22)
> 
> Results :
> 
> Tests in error: 
>   » ParameterizedAssertion test(ids[1])
> 
> Tests run: 1, Failures: 0, Errors: 1, Skipped: 0

I expect: 
failures: 1, errors: 0

--- a/src/main/java/org/junit/experimental/theories/internal/ParameterizedAssertionError.java
+++ b/src/main/java/org/junit/experimental/theories/internal/ParameterizedAssertionError.java
@@ -4,13 +4,13 @@ import java.util.Arrays;
 import java.util.Collection;
 import java.util.Iterator;
 
-public class ParameterizedAssertionError extends RuntimeException {
+public class ParameterizedAssertionError extends AssertionError {
     private static final long serialVersionUID = 1L;
 
     public ParameterizedAssertionError(Throwable targetException,
             String methodName, Object... params) {
-        super(String.format("%s(%s)", methodName, join(", ", params)),
-                targetException);
+        super(String.format("%s(%s)", methodName, join(", ", params)));
+        this.initCause(targetException);
     }
 
     @Override

